File: python/2085.count-common-words-with-one-occurrence.py
from collections import Counter
from typing import List


class Solution:
    def countWords(self, words1: List[str], words2: List[str]) -> int:
        set1: dict[str, int] = Counter(words1)
        set2: dict[str, int] = Counter(words2)

        res = 0

        for w in set1.keys():
            if w in set2 and set1[w] == 1 and set2[w] == 1:
                res += 1
                set1[w] = 0
                set2[w] = 0

        for w in set2.keys():
            if w in set1 and set1[w] == 1 and set2[w] == 1:
                res += 1

        return res

File: cpp/2085.count-common-words-with-one-occurrence.cpp
#include <string>
#include <unordered_map>
#include <vector>

using std::string;
using std::unordered_map;
using std::vector;

class Solution {
public:
  int countWords(vector<string> &words1, vector<string> &words2) {
    unordered_map<string, int> m1;
    unordered_map<string, int> m2;

    for (string &w : words1) {
      ++m1[w];
    }

    for (string &w : words2) {
      ++m2[w];
    }

    int res = 0;

    for (string &w : words1) {
      if (m1[w] == 1 && m2[w] == 1) {
        ++res;
        m1[w] = 0;
        m2[w] = 0;
      }
    }

    for (string &w : words2) {
      if (m1[w] == 1 && m2[w] == 1) {
        ++res;
      }
    }

    return res;
  }
};